### PR TITLE
Release 1.17.0 (Initial public preview release of pandas API)

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
Release 1.17.0 (Initial public preview release of pandas API)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,14 +59,18 @@
 - Improved error experience of `Session.write_pandas` method that `NotImplementError` will be raised when called.
 - Aligned error experience with reusing a closed session in non-local execution.
 
-## 1.16.0 (TBD)
+## 1.16.0 (2024-05-07)
```
Comment (sfc-gh-azhan) [inline CHANGELOG.md:62]: @sfc-gh-aling @sfc-gh-mabrennan those changes are made to sync with 1.16.0 release note: https://github.com/snowflakedb/snowpark-python/releases/tag/v1.16.0

File: src/snowflake/snowpark/modin/plugin/PANDAS_CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-## 1.16.0a1 (tbd)
+## 1.17.0 (2024-05-21)
```
Comment (sfc-gh-azhan) [inline src/snowflake/snowpark/modin/plugin/PANDAS_CHANGELOG.md:1]: @sfc-gh-nkrishna change this too to make sure we know what the cut is.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
   - `DataFrame.create_or_replace_temp_view`
   - `DataFrame.create_or_replace_dynamic_table`
 
-### Improvements
+#### Improvements
 
 - Improved error message to remind users set `{"infer_schema": True}` when reading csv file without specifying its schema.
 
-### Local Testing Updates
+### Snowpark pandas API updates
```
Comment (sfc-gh-lspiegelberg) [inline CHANGELOG.md:19]: nit: here it is `updates` but in the other heading it is `Updates`. Maybe let's stick with one?

File: CHANGELOG.md [hunk 4]
```diff
@@ -74,6 +86,11 @@
 
 - Fixed a bug that caused NaT and NaN values to not be recognized.
```
Comment (sfc-gh-lspiegelberg) [inline CHANGELOG.md:87]: nit: `NaT` and `NaN` (in ticks)?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,20 +1,28 @@
...
+#### Improvements
 
 - Improved error message to remind users set `{"infer_schema": True}` when reading csv file without specifying its schema.
 
-### Local Testing Updates
+### Snowpark pandas API Updates
+
+#### New Features
+
+- Initial public preview release of Snowpark pandas API.
```
Comment (sfc-gh-lspiegelberg) [inline CHANGELOG.md:23]: nit: Should we maybe add a short description on what the Snowpark pandas API is? Like 1-2 lines max, because it's a changelog.

File: CHANGELOG.md [hunk 6]
```diff
@@ -72,8 +84,13 @@
 
 #### Bug Fixes
 
-- Fixed a bug that caused NaT and NaN values to not be recognized.
+- Fixed a bug that caused `NaT` and `NaN` values to not be recognized.
 - Fixed a bug when inferring schema, single quotes are added to stage files already have single quotes.
+- Fixed a bug where DataFrameReader.csv was unable to handle quoted values containing a delimiter.
```
Comment (sfc-gh-lspiegelberg) [inline CHANGELOG.md:89]: nit: `DataFrameReader.csv`

**Review by sfc-gh-lspiegelberg — APPROVED** [review]
PuPr! Great to see this milestone be achieved 🍾 !

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,20 +1,28 @@
 # Release History
 
-## 1.17.0 (TBD)
+## 1.17.0 (2024-05-21)
 
-### New Features
+### Snowpark Python API Updates
+
+#### New Features
```
Comment (sfc-gh-helmeleegy) [inline CHANGELOG.md:7]: Can we merge the two "New Features" sections?

Comment (sfc-gh-azhan) [reply CHANGELOG.md:7]: This new features is only used for python API.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,20 +1,28 @@
 # Release History
 
-## 1.17.0 (TBD)
+## 1.17.0 (2024-05-21)
```
[the remark this answers is not in the document]
Comment (sfc-gh-azhan) [reply CHANGELOG.md:3]: This is the release cut and will be used in the release note. We don't want to show the TBD version here right?

File: CHANGELOG.md (hunk 5 above)
Comment (sfc-gh-azhan) [reply CHANGELOG.md:23]: I didn't add more based on what we did for prpr and pupr in both python and pandas API. @sfc-gh-dolee @sfc-gh-mabrennan Let me know if you want to rephrase here.

File: CHANGELOG.md [hunk 9]
```diff
@@ -72,8 +79,13 @@
 
 #### Bug Fixes
 
-- Fixed a bug that caused NaT and NaN values to not be recognized.
+- Fixed a bug that caused `NaT` and `NaN` values to not be recognized.
 - Fixed a bug when inferring schema, single quotes are added to stage files already have single quotes.
+- Fixed a bug where `DataFrameReader.csv` was unable to handle quoted values containing a delimiter.
+- Fixed a bug that when there is `None` value in an arithmetic calculation, the output should remain `None` instead of `math.nan`.
+- Fixed a bug in function `sum` and `covar_pop` that when there is `math.nan` in the data, the output should also be `math.nan`.
+- Fixed a bug that stage operation can not handle directories.
```
Comment (sfc-gh-nkrishna) [inline CHANGELOG.md:87]: Why these changes?

File: CHANGELOG.md [hunk 10]
```diff
@@ -51,14 +54,18 @@
...
 
 ### New Features
 
 - Support stored procedure register with packages given as Python modules.
 - Added snowflake.snowpark.Session.lineage.trace to explore data lineage of snowfake objects.
 - Added support for structured type schema parsing.
 
+### Bug Fixes
+
+- Fixed a bug when inferring schema, single quotes are added to stage files already have single quotes.
```
[the remark this answers is not in the document]
Comment (sfc-gh-azhan) [reply CHANGELOG.md:67]: This is a fix for the previous release which didn't update the change log correctly after release.

File: CHANGELOG.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (sfc-gh-azhan) [reply CHANGELOG.md:3]: @sfc-gh-aling should we add new release version TBD here? If yes, what's the next version?

File: CHANGELOG.md (hunk 7 above)
Comment (sfc-gh-helmeleegy) [reply CHANGELOG.md:7]: Oh, that makes sense. Sorry didn't notice that these headers are nested separately under `Snowpark Python API Updates` and `Snowpark pandas API Updates`. Ok, great!

File: CHANGELOG.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (sfc-gh-helmeleegy) [reply CHANGELOG.md:3]: I think it's ok not to include it in this PR. But we will still need to know what release version the following PRs will need to use.

**Review by sfc-gh-dolee — APPROVED** [review]

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,20 +1,28 @@
...
 
-## 1.17.0 (TBD)
+## 1.17.0 (2024-05-21)
 
-### New Features
+### Snowpark Python API Updates
+
+#### New Features
 
 - Added support to add a comment on tables and views using functions listed below:
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:9]: add "the" before "functions listed below"

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,20 +1,28 @@
...
   - `DataFrameWriter.save_as_table`
   - `DataFrame.create_or_replace_view`
   - `DataFrame.create_or_replace_temp_view`
   - `DataFrame.create_or_replace_dynamic_table`
 
-### Improvements
+#### Improvements
 
 - Improved error message to remind users set `{"infer_schema": True}` when reading csv file without specifying its schema.
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:17]: Add "to" before "set".
csv -> CSV

File: CHANGELOG.md (hunk 5 above)
Comment (sfc-gh-mabrennan) [reply CHANGELOG.md:23]: The feature will have a release note in the real Snowflake release notes. I don't think we should add descriptions here. If you want to add a link to the documentation, go ahead.

File: CHANGELOG.md [hunk 13]
```diff
@@ -32,12 +40,7 @@
 
 #### Bug Fixes
 
-- Fixed a bug in DataFrameReader.csv unable to handle quoted values containing delimiter.
-- Fixed a bug that when there is `None` value in arithmetic calculation, the output should remain `None` instead of `math.nan`.
-- Fixed a bug in function `sum` and `covar_pop` that when there is `math.nan` in the data, the output should also be `math.nan`.
-- Fixed a bug that stage operation can not handle directories.
-- Fixed a bug that `DataFrame.to_pandas` should take Snowflake numeric types with precision 38 as `int64`.
-- Fixed a bug that stored proc and udf should not remove imports already in the sys.path during the clean-up step.
+- Fixed a bug that stored proc and udf should not remove imports already in the `sys.path` during the clean-up step.
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:43]: stored proc -> stored procedure
udf -> UDF

File: CHANGELOG.md [hunk 14]
```diff
@@ -32,12 +40,7 @@
...
 - Fixed a bug that when processing datetime format, fractional second part is not handled properly.
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:44]: add "the" before "fractional second part".

File: CHANGELOG.md (hunk 10 above)
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:67]: Fixed a bug where, when inferring a schema, single quotes were added to stage files that already had single quotes.

**Review by sfc-gh-mabrennan — APPROVED** [review]
Thank you. I made a few small suggestions.

**Review by sfc-gh-helmeleegy — APPROVED** [review]

**Review by sfc-gh-aling — APPROVED** [review]
LGTM, thanks for help cleaning up the changelog for local testing!

**Review by sfc-gh-nkrishna — APPROVED** [review]
Thanks for cleaning up the CHANGELOG!
